packages/amplify-category-geo/src/service-walkthroughs/mapWalkthrough.ts

```diff
@@ -0,0 +1,203 @@
+    context.print.info('- Map pricing plan (default: RequestBasedUsage)');
+    context.print.info('');
+
+    if(await context.amplify.confirmPrompt('Do you want to configure advanced settings?', false)) {
+        // get the map style parameters
+        parameters = merge(parameters, await mapStyleWalkthrough(parameters));
+
+        // get the pricing plan
+        parameters = merge(parameters, await pricingPlanWalkthrough(parameters, ServiceName.Map));
+    }
+    else {
+        parameters.dataProvider = DataProvider.Esri;
+        parameters.mapStyleType = EsriMapStyleType.Streets;
+        parameters.pricingPlan = PricingPlan.RequestBasedUsage;
+    }
+
+    return parameters;
+};
+
+export const mapStyleWalkthrough = async (parameters: Partial<MapParameters>): Promise<Partial<MapParameters>> => {
+    const mapStyleChoices = [
+        { name: 'Streets (data provided by Esri)', value: MapStyle.VectorEsriStreets },
+        { name: 'Berlin (data provided by Here)', value: MapStyle.VectorHereBerlin },
+        { name: 'Topographic (data provided by Esri)', value: MapStyle.VectorEsriTopographic },
+        { name: 'Navigation (data provided by Esri)', value: MapStyle.VectorEsriNavigation },
+        { name: 'LightGrayCanvas (data provided by Esri)', value: MapStyle.VectorEsriLightGrayCanvas },
+        { name: 'DarkGrayCanvas (data provided by Esri)', value: MapStyle.VectorEsriDarkGrayCanvas }
+    ];
+    const mapStyleDefault = parameters.dataProvider && parameters.mapStyleType ?
+        getGeoMapStyle(parameters.dataProvider, parameters.mapStyleType) : 'VectorEsriStreets';
+
+    const mapStyleTypePrompt = {
+        type: 'list',
+        name: 'mapStyle',
+        message: `Specify the map style. Refer ${apiDocs.mapStyles}`,
+        choices: mapStyleChoices,
+        default: mapStyleDefault
+    };
+    return getMapStyleComponents((await inquirer.prompt([mapStyleTypePrompt])).mapStyle as MapStyle);
+};
+
+/**
+ * Starting point for CLI walkthrough that updates an existing map resource
+ * @param context The Amplify Context object
+ * @param parameters The configurations of the Map resource
+ * @param resourceToUpdate Name of the Map resource to update
+ */
+export const updateMapWalkthrough = async (
+    context: $TSContext,
+    parameters: Partial<MapParameters>,
+    resourceToUpdate?: string
+): Promise<Partial<MapParameters>> => {
+    const mapResources = ((await context.amplify.getResourceStatus()).allResources as any[])
+    .filter(resource => resource.service === ServiceName.Map)
+
+    if (mapResources.length === 0) {
+        context.print.error('No Map resource to update. Use "amplify add geo" to create a new Map.');
+        return parameters;
+    }
+
+    const mapResourceNames = mapResources.map(resource => resource.resourceName);
+
+    if (resourceToUpdate) {
+        if (!mapResourceNames.includes(resourceToUpdate)) {
```

Comment:
these 2 conditions can be and-ed

Reply:
The behavior here is that if there is a valid `resourceToUpdate` input then we need to skip the `resourceQuestion` in the `else` statement below. If we put an `&&` operation, it would hit the `else` statement and asks `resourceQuestion`.